[Saravanan Sivaram Jeychand] iP

sivaramjeychand opened the pull request:
# 🚀 HamiltonianGuy - Task Manager Chatbot

> _"Your mind is for having ideas, not holding them." – David Allen ([source](https://gettingthingsdone.com))_

HamiltonianGuy is a **simple, text-based chatbot** that helps users manage tasks efficiently. It's:

- **Fast** ⚡
- **Simple to use** 📝
- **Persistent (saves tasks automatically)** 💾

## 📌 Features Implemented
1. **Basic Commands**
   - `todo <task>` – Adds a ToDo task.
   - `deadline <task> /by <date>` – Adds a task with a deadline.
   - `event <task> /from <start> /to <end>` – Adds an event.

2. **Task Management**
   - ✅ Mark tasks as done (`mark <index>`)
   - ❌ Unmark tasks (`unmark <index>`)
   - 🗑 Delete tasks (`delete <index>`)

3. **Advanced Features**
   - 🔍 `find <keyword>` – Search for tasks by keyword.
   - 💾 **Auto-save** tasks to a file.
   - 📆 **Date & Time Handling** with `LocalDateTime`.

## 🛠 Setup Instructions
```sh
# Compile and Run
./gradlew build
java -jar build/libs/Hamiltonianguy-1.0.jar

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

sivaramjeychand committed:
Level-0: Rename, Greet, Exit

sivaramjeychand committed:
Level-1: Echo

sivaramjeychand committed:
Level-2: Add, List

sivaramjeychand committed:
Level-3: Mark as Done

sivaramjeychand committed:
Level-3: Mark as Done (minor changes)

sivaramjeychand committed:
Level-4: ToDos, Events, Deadlines
Refactored the code to follow better OOP principles and added more classes. But I still am not doing the toString method properly. Might use StringBuilder later on.

sivaramjeychand committed:
Level-5,6: Handle errors and delete

Added try catch blocks and specific messages for each exception. Added the delete function in this commit as well, as it was just one extra method, and I was already using ArrayList to store tasks from the very beginning. Rewrote and shuffled around code in the main class to handle all the various exceptions. With how late I started this I am happy with my progress. I should add more flavour and personality in the upcoming weeks.

sivaramjeychand committed:
Implement Level-7, Add Storage class.

sivaramjeychand committed:
Implement Level-7: Add Storage class.

sivaramjeychand committed:
Implement Level-8: Add Date and TIme functionality

sivaramjeychand committed:
Refactor into packages.

sivaramjeychand committed:
Merge branch 'add-gradle-support'

sivaramjeychand committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	data/HamiltonianGuy.txt
#	src/main/java/DeadlineTask.java
#	src/main/java/EventTask.java
#	src/main/java/ToDoTask.java
#	src/main/java/hamiltonianguy/TaskList.java
#	src/main/java/hamiltonianguy/tasks/Task.java

sivaramjeychand committed:
Add Javadocs

sivaramjeychand committed:
Merge branch 'branch-A-JavaDoc'

sivaramjeychand committed:
Change some spacing to comply with coding standard

sivaramjeychand committed:
Merge branch 'branch-A-CodingStandard'

sivaramjeychand committed:
Resolve merge conflicts and add gradle

sivaramjeychand committed:
Resolve merge conflicts in hamiltonianguy.HamiltonianGuy

sivaramjeychand committed:
Add save command and handle save command to main class and TaskListt

sivaramjeychand committed:
Package app as JAR file. Change build.gradle to produce a JAR file.

sivaramjeychand committed:
Add Test classes to utilise JUnit testing

sivaramjeychand committed:
Minor edits to build.gradle

sivaramjeychand committed:
Start work on the UI
Add 2 classes

sivaramjeychand committed:
Add JavaFx and a simple GUI to the chatbot

OvOtter commented on data/HamiltonianGuy.txt:
```diff
@@ -0,0 +1,4 @@
+D | 1 | return book | 2024-02-01 1800
```
Save files should not be uploaded to github (put in gitignore!)

OvOtter commented on src/main/java/hamiltonianguy/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+     */
+
+    public String getTaskListAsString() {
+        StringBuilder sb = new StringBuilder();
```
sb is not a concrete variable name. Maybe use taskListString?

OvOtter commented on src/main/java/hamiltonianguy/TaskList.java:
```diff
@@ -0,0 +1,153 @@
+        System.out.println("______________________________________");
+        System.out.println("Here are the matching tasks in your list:");
+
+        boolean found = false;
```
Maybe you can change to isFound

OvOtter reviewed (commented):
lgtm :)

sivaramjeychand committed:
feat: Implement GUI chat interface and task search

- Add JavaFX-based GUI for HamiltonianGuy chatbot
- Implement chat UI with user messages right-aligned and bot responses left-aligned
- Show a welcome message with a logo on startup
- Implement `find` command to allow users to search tasks by keyword
- Integrate JavaFX for structured task display using `VBox`
- Configure Gradle to include `resources/` for loading assets
- Fix image loading issues and verify proper message alignment
- Package project as a Gradle build with correct dependencies

sivaramjeychand committed:
feat: Add assertions to ensure data integrity and correctness

- Add assertions in `TaskList` to validate task index before deletion
- Add assertions in `Task` to prevent null or empty task descriptions
- Add assertions in `Storage` to ensure directory exists before saving
- Add assertions in `HamiltonianGuy` to validate user input before processing
- Enable assertions in Gradle (`-ea` flag) for development testing

This improves code reliability by catching invalid states early

sivaramjeychand committed:
Remove all wildcard imports

sivaramjeychand committed:
Merge pull request #1 from sivaramjeychand/branch-A-Assertions

Branch a assertions

sivaramjeychand committed:
Merge pull request #2 from sivaramjeychand/master

merge master to A-CodeQuality

sivaramjeychand committed:
Add sort method for deadline tasks to sort by deadline in chronological order

sivaramjeychand committed:
Merge remote-tracking branch 'origin/branch-A-CodeQuality'

sivaramjeychand committed:
Add some more GUI features like user and bot avatars and fix bug with delete command

sivaramjeychand committed:
Add GFMD to README to create a GitHub Page

sivaramjeychand committed:
Update README anf fix some minor errors

sivaramjeychand committed:
Edit gitignore to ensure the save file is not committed

sivaramjeychand committed:
Remove wildcard imports in main class

sivaramjeychand committed:
Remove some comments and clean up code
